usr/src/uts/common/fs/zfs/dsl_scan.c

```diff
@@ -69,13 +69,14 @@ int zfs_scan_idle = 50;			/* idle window in clock ticks */
 
 int zfs_scan_min_time_ms = 1000; /* min millisecs to scrub per txg */
 int zfs_free_min_time_ms = 1000; /* min millisecs to free per txg */
+int zfs_obsolete_min_time_ms = 500; /* min millisecs to obsolete per txg */
 int zfs_resilver_min_time_ms = 3000; /* min millisecs to resilver per txg */
 boolean_t zfs_no_scrub_io = B_FALSE; /* set to disable scrub i/o */
 boolean_t zfs_no_scrub_prefetch = B_FALSE; /* set to disable scrub prefetch */
 enum ddt_class zfs_scrub_ddt_class_max = DDT_CLASS_DUPLICATE;
 int dsl_scan_delay_completion = B_FALSE; /* set to delay scan completion */
 /* max number of blocks to free in a single TXG */
-uint64_t zfs_free_max_blocks = UINT64_MAX;
+uint64_t zfs_async_block_max_blocks = UINT64_MAX;
```

Comment:
Does this name change really belong in the device evacuation patch?

Reply:
It is relevant because we've changed the meaning to cover not only async frees but also obsoleting of the indirect mapping - see dsl_scan_obsolete_block_cb().  The process is similar to background deletion of snapshots.  Now, the name of the new variable might leave something to be desired...